pkg/network/setup/netstat.go

```diff
@@ -167,20 +169,20 @@ func ifacesStatusFromGuestAgent(vmiIfacesStatus []v1.VirtualMachineInstanceNetwo
 	for _, guestAgentInterface := range guestAgentInterfaces {
 		if vmiIfaceStatus := netvmispec.LookupInterfaceStatusByMac(vmiIfacesStatus, guestAgentInterface.Mac); vmiIfaceStatus != nil {
 			updateVMIIfaceStatusWithGuestAgentData(vmiIfaceStatus, guestAgentInterface, vmiInterfacesSpecByName)
+			vmiIfaceStatus.InfoSource = netvmispec.InfoSourceDomainAndGA
```

Comment:
The backward comparability with older virt-launchers seems to be here.
Suggested fix is to condition this per the `Name` field value, from the `guestAgentInterface`.

Older virt-launchers will have this non-empty (due to the old merge).
New ones will have nothing here, as `Name` is no longer in use.

* Note: The `InfoSource` exists for only a short period, therefore it is suggested to ignore the case where it was available and not treat it as a special case.

Reply:
End result should be:
- For new launchers, set the source.
- For old launcher, set it to empty (as we do not know the origin really).

Reply:
The name will be there since you still call the merge internally. I think the merge is redundant now.
EDIT: ignore, I see it is done in the next commit.
The backward compatibility plan sounds good.